### ISSUE TITLE
redundance data preparation cycle even with dataset.load()

### ISSUE THREAD

**Issue by thegodone** [aimat-lab/gcnn_keras#74]
I don't understand why even with the load function I still see this process:
![image](https://user-images.githubusercontent.com/1186658/188324055-d5fb0871-9e43-49e2-b237-1f11e4345400.png)

![image](https://user-images.githubusercontent.com/1186658/188324197-92508de4-5bcd-45dc-88a0-ef925b248c22.png)
![image](https://user-images.githubusercontent.com/1186658/188324216-d77246f4-696a-4efb-8258-646e4239696d.png)

![image](https://user-images.githubusercontent.com/1186658/188324256-a6c091f2-fd0c-4dfb-89f7-96aa67130f93.png)

is there something in the HyperParameter that can cause this ?

auxilary question how to avoid the Openbabel warning we not use it in OCHEM.

**Comment by PatReis** [comment]
It should better say "" ... process molecules 0 from 4851". 
But the `read_in_memory` and `set_attributes` functions were doing different things in earlier versions. Now they are actually more or less redundant. I changed it and made them alias or synonyms.

You can now leave out the ``set_attributes`` from hyper-parameter. 

However, if the model wants a different graph, than which is read by default (e.g. different attributes or directed or with hydrogen etc.) then I see no other way than redoing the read process. 

By the way, you can now read the mol-information separately with privat `mol_list = MoleculeNetDataset._read_in_memory_mol_blocks()`.

I removed the warning. It will raise just a standard python import error if it comes to it.

**Comment by PatReis** [comment]
For the small datasets I see no harm in loading twice if hyperparameter of a model wants to ensure a certain read and graph generation. But if you don't like it you can now just leave out the set_attributes in hyper-parameter and controll the `read_in_memory` by yourself.

**Comment by thegodone** [comment]
I don't know what is the best / simpler way to do so, I guess if overwrite = True it means I want to redo all the process even recreate sdf etc..., but if no overwrite it means for me I have a good data already process for my model seetings and I want just to redo the model (like Optuna HPO) or adding more epochs etc... so nothing about dataprocess but more like to model improvement / finetune

**Comment by PatReis** [comment]
Yes, that should work now like this. Whether you want it in hyperparamters or do it manually in your code is up to you. Just remove one.

**Comment by PatReis** [comment]
The `prepare_data` was always done only once, as you said, unless overwrite = True. This is the time consuming part, translating smiles and guessing conformers.
